pkg/providers/amifamily/resolver.go

```diff
@@ -88,6 +95,22 @@ type FeatureFlags struct {
 // DefaultFamily provides default values for AMIFamilies that compose it
 type DefaultFamily struct{}
 
+func (d *DefaultFamily) FetchAMIsFromSSM(ctx context.Context, ssmQuery string, ssmCache *cache.Cache, ssmapi ssmiface.SSMAPI, cm *pretty.ChangeMonitor) (string, error) {
```

Comment:
I think I see your point now with where to store the cache/API/etc. I could still see us creating a separate SSM provider, but I think you're right that we need a custom data type to return from the AMIFamily and the resolver can do the actual SSM resolution logic since that's common among the interfaces. Seems good to revert back to the way you had it before

Reply:
The other thing that you could do here is you could create an SSM provider and pass it through from the Resolver like 

```go
// Resolver is able to fill-in dynamic launch template parameters
type Resolver struct {
	amiProvider *Provider
	ssmProvider *ssm.Provider
}
```

```go
func GetAMIFamily(amiFamily *string, options *Options) AMIFamily {
	switch aws.StringValue(amiFamily) {
	case v1alpha1.AMIFamilyBottlerocket:
		return &Bottlerocket{Options: options, SSMProvider: ssmProvider}
	case v1alpha1.AMIFamilyUbuntu:
		return &Ubuntu{Options: options, SSMProvider: ssmProvider}
	case v1alpha1.AMIFamilyCustom:
		return &Custom{Options: options}
	default:
		return &AL2{Options: options, SSMProvider: ssmProvider}
	}
}
```